### DISCUSSION TITLE
How to inject an envvar into a formula?

### DISCUSSION THREAD

**Discussion by Cerebus** [Homebrew/discussions#4877] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.18-33-g83b4b2f
ORIGIN: https://github.com/Homebrew/brew
HEAD: 83b4b2f2f4b28e1f035e88ac41140f1c8b0c9c9b
Last commit: 23 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 953a265a52e62627b94587235e645f13ce3b2b59
Core tap last commit: 3 hours ago
Core tap branch: master
Core tap JSON: 01 Nov 17:44 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: emacsclient
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.42.0 => /usr/local/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.6-x86_64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

My dev environment provides a go package proxy and blocks (some) upstreams package repos.  I have GOPROXY set in my global config (~/.config/go/env) and can verify with `go env`.  However, go tool formulae break b/c (AFAICT) it's not picking up the envvar.    `bin/brew` is filtering the shell environment, so I can't just set it explicitly.  Is there a workaround or an option I'm missing here?

**Comment by alebcay** [comment]
Yes, `brew` filters out most environment variables, see https://docs.brew.sh/Formula-Cookbook#using-environment-variables.

There is a specific list of variables allowed that is defined in `bin/brew`; also, any environment variable starting with `HOMEBREW_` should be allowed through. The formula may need to be edited locally so that inside the build environment, the value of e.g. `HOMEBREW_GOPROXY` is copied into `GOPROXY`.
